### PR TITLE
feat: more selectors

### PR DESCRIPTION
**Pull request by hacdias**
This changes a bit the API so you need to call the bundler as a function to make it. ~This is because the `ready` variable is now wrapped in a function instead of in the state. Why? This way, this bundle will be compatible with [`createCacheBundle`](https://reduxbundler.com/api/included-bundles.html#asynccountbundle).~ API Address is stored.

Also, added the selectors: ```selectIpfsApiAddress``` and ```selectIpfsInitFailed```.

### CODE REVIEW

**Comment by hacdias** [comment]
Could you merge and release a new version, please? 😄

File: index.js [hunk 1]
```diff
@@ -3,55 +3,82 @@ const getIpfs = require('window.ipfs-fallback')
...
+        return dispatch({ type: 'IPFS_INIT_FAILED', error })
+      }
+
+      dispatch({ type: 'IPFS_INIT_FINISHED', payload: identity })
+    },
+
+    doUpdateIpfsAPIAddress: (apiAddress) => ({dispatch, store}) => {
+      // root.ipfs needs to be set to null so getIpfs doesn't
+      // use the previous connection.
+      root.ipfs = null
```
Comment (olizilla) [inline index.js]: `root.ipfs` could be the `window.ipfs`  instance from companion. Now this bundle exports a function, we should store the current ipfs instance as a variable in the function scope, and not mess with the `root.ipfs` instance.

**Comment by hacdias** [comment]
@olizilla just made the updates as you suggested 😄

File: index.js [hunk 2]
```diff
@@ -1,57 +1,80 @@
...
   apiAddress: '/ip4/127.0.0.1/tcp/5001',
-  ipfsReady: false
+  identity: null,
+  error: null
 }
 
-module.exports = {
-  name: 'ipfs',
+module.exports = function () {
+  let ready = false
```
[the remark this answers is not in the document]
Comment (hacdias) [reply index.js]: Hmm, we can do that, but that way it won't be compatible with `createCacheBundler`. How do you suggest to persist the apiAddress?

**Comment by hacdias** [comment]
@olizilla I'm now using the package [`store`](https://www.npmjs.com/package/store) to save the API Address

**Comment by hacdias** [comment]
I believe all issues are fixed now 😄 

Btw, could you add write permission for @ipfs-shipyard/gui on this repo?

File: README.md [hunk 3]
```diff
@@ -56,14 +56,26 @@ Adds the following methods to the redux store
 
 - `boolean` - Is the IPFS instance ready to use yet?
 
+#### `store.selectIpfsInitFailed()`
+
+- `boolean` - Did the IPFS instance failed to start?
```
Comment (olizilla) [inline README.md]: Did the IPFS instance failed to start?
-> 
Did the IPFS instance fail to start?

**Pull request merged**